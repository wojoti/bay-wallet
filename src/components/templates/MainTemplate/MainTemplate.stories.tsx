import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {MarketData} from 'data/index';

import MainTemplate from './MainTemplate';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'Templates/MainTemplate',
  component: MainTemplate,
  argTypes: {
    onNotificationPress: {action: 'onNotificationPress'},
    onUserProfilePress: {action: 'onUserProfilePress'},
    onDetailsPress: {action: 'onDetailsPress'},
    onTransferPress: {action: 'onTransferPress'},
    onWithdrawPress: {action: 'onWithdrawPress'},
    onViewMorePress: {action: 'onViewMorePress'},
  },
} as ComponentMeta<typeof MainTemplate>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof MainTemplate> = args => (
  <MainTemplate {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  data: MarketData,
};
