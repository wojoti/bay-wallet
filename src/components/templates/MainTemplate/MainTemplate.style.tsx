import {StyleSheet} from 'react-native';
import {primary, secondary} from 'theme/colors';

const style = StyleSheet.create({
  wrapper: {
    backgroundColor: secondary,
    height: '100%',
  },
  contentWithRadius: {
    backgroundColor: primary,
    borderWidth: 0,
    borderColor: 'transparent',
    borderBottomLeftRadius: 30,
    borderBottomRightRadius: 30,
  },
  topContent: {marginTop: 60, width: '85%'},
  balanceBoxSizer: {
    width: '93%',
    marginTop: 35,
  },
  width: {width: '85%'},
});
export default style;
