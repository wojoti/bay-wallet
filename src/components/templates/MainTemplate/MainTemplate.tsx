import {Container, Label} from 'components/atoms';
import {MainUserArea} from 'components/organisms';
import {PropsWithChildren} from 'react';
import style from './MainTemplate.style';

export type MainTemplateProps = PropsWithChildren<{
  onNotificationPress: () => void;
  onUserProfilePress: () => void;
  onDetailsPress: () => void;
  onTransferPress: () => void;
  onWithdrawPress: () => void;
  testId?: string;
}>;

const MainTemplate = ({
  onNotificationPress,
  onUserProfilePress,
  onDetailsPress,
  onTransferPress,
  onWithdrawPress,
  testId,
}: MainTemplateProps) => (
  <Container customStyle={style.wrapper} testId={testId}>
    <Container flex={1}>
      <MainUserArea
        onNotificationPress={onNotificationPress}
        onUserProfilePress={onUserProfilePress}
        onDetailsPress={onDetailsPress}
        onTransferPress={onTransferPress}
        onWithdrawPress={onWithdrawPress}
      />
    </Container>
    <Container flex={1}>
      <Label>Latest market</Label>
    </Container>
  </Container>
);

export default MainTemplate;
