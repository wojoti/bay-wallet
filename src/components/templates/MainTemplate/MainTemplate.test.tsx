import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import MainTemplate, {MainTemplateProps} from './MainTemplate';

const props: MainTemplateProps = {
  testId: 'test-maintemplate-id',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<MainTemplate {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render maintemplate', () => {
  render(<MainTemplate {...props} />);
  const maintemplateElement = screen.getByTestId('test-maintemplate-id');
  expect(maintemplateElement).toBeOnTheScreen();
});
