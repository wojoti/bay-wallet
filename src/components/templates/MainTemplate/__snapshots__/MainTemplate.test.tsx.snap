// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "stretch",
        "flex": undefined,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "width": "100%",
      },
      {
        "backgroundColor": "white",
        "height": "100%",
      },
    ]
  }
  testID="test-maintemplate-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "backgroundColor": "#363062",
          "borderBottomLeftRadius": 30,
          "borderBottomRightRadius": 30,
          "borderColor": "transparent",
          "borderWidth": 0,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "marginTop": 60,
            "width": "85%",
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "alignItems": "stretch",
              "flex": undefined,
              "flexDirection": "row",
              "justifyContent": "flex-start",
              "width": "100%",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "flex-start",
                "flex": 2.6,
                "flexDirection": "column",
                "justifyContent": "flex-start",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "Poppins-Light",
                "fontSize": 14,
                "fontWeight": "400",
                "textAlign": "left",
              }
            }
            testID="userbar-welcome"
          >
            Good Morning
          </Text>
          <Text
            style={
              {
                "color": "white",
                "fontFamily": "Poppins-Bold",
                "fontSize": 24,
                "textAlign": "left",
              }
            }
            testID="userbar-username"
          >
            David Kowalski
          </Text>
        </View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "width": "100%",
              },
              undefined,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#363062",
                "borderColor": "#363062",
                "borderRadius": 8,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": 65,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID="userbar-notification"
          >
            <SvgMock
              height={26}
              width={24}
            />
          </View>
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#363062",
                "borderColor": "#363062",
                "borderRadius": 8,
                "borderStyle": "solid",
                "borderWidth": 1,
                "height": 65,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
            testID="userbar-userprofile"
          >
            <SvgMock
              height={40}
              width={40}
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
    >
      Latest market
    </Text>
  </View>
</View>
`;
